src/reporter.js

```diff
@@ -111,17 +112,21 @@ function createTestRecord(test, specRelativePath) {
     testName = testFullName.replace(className+' -- ','');
   }
 
-  if (test.state === 'failed') {
-    var err = test.err;
-    var failure = {$: {message: err.message, type: err.name}, _: err.stack}; // Note, to force CDATA add "<< "
-    const unsafeRegex = /[^ A-Za-z0-9._-]/g;
-    var imageBasename = testFullName.replaceAll(unsafeRegex, '').substring(0, 242)+' (failed).png';
-    var imageFile = path.join(config.screenshotsFolder, specRelativePath, imageBasename);
-    var imageScreenshot = '[[ATTACHMENT|'+imageFile+']]';
-    return {$: {name: testName, classname: className, time: test.duration/1000}, failure: failure, 'system-out': imageScreenshot};
-  } else {
-    return {$: {name: testName, classname: className, time: test.duration/1000}};
+  const unsafeRegex = /[^ A-Za-z0-9._-]/g;
+  switch (test.state) {
+    case 'failed':
+      var err = test.err;
+      var failure = {$: {message: err.message, type: err.name}, _: err.stack}; // Note, to force CDATA add "<< "
+      var imageBasename = testFullName.replaceAll(unsafeRegex, '').substring(0, 242)+' (failed).png';
+      var imageFile = path.join(config.screenshotsFolder, specRelativePath, imageBasename);
+      var imageScreenshot = '[[ATTACHMENT|'+imageFile+']]';
+      return {$: {name: testName, classname: className, time: test.duration/1000}, failure: failure, 'system-out': imageScreenshot};
+    case 'pending':
+      return {$: {name: testName, classname: className, time: test.duration/1000}, skipped: null};
+    default:
+      return {$: {name: testName, classname: className, time: test.duration/1000}};
```

Comment:
I think it is better (no  repetition) written as:
```
var record = {$: {name: testName, classname: className, time: test.duration/1000}};
switch (test.state) {
  case 'failed':
    record['failure'] = {$: {message: err.message, type: err.name}, _: err.stack};
    const unsafeRegex = /[^ A-Za-z0-9._-]/g;
    let imageBasename = testFullName.replaceAll(unsafeRegex, '').substring(0, 242)+' (failed).png';
    let imageFile = path.join(config.screenshotsFolder, specRelativePath, imageBasename);
    if (fs.existsSync(imageFile) {
      record['system-out'] = '[[ATTACHMENT|'+imageFile+']]';
    }
    break;
  case 'pending':
    record['system-out'] = nil;
    break;
}

return record;

Reply:
The suggestions look good, but have a few comments:
- The `unsafeRegex` generates a **lint error** when within the switch statement
  ```
  error  Unexpected lexical declaration in case block  no-case-declarations
  ```
- Testing for the actual image, and if **not found** nothing will indicate that anything went wrong. Having the annotation show a missing file can be helpful. For example:
 
  ![image](https://github.com/testspace-com/cypress-xml-reporter/assets/1229367/8cef67f0-1c30-4557-93c0-c8980095e72d)

Reply:
The error annotation is not helpful to the user - it is confusing!

For you (the maintainer) it might be helpful to log-debug...

Reply:
> The unsafeRegex generates a lint error when within the switch statement

Just FYI, similar to C/C++, `switch` in JS may not have `break` - the individual `case` are not "blocks" and anything defined in one is visible in the following ones... In general, I think, `var`/`let` should also be caught by the linter.

Anyway, keep `unsafeRegex` outside the "switch". However, I would recommend if you rename it `UNSAFE_REGEX` (as it is a literal constant)

Reply:
Note, using `let` within the case would require using a block:
```
case 'failed': {
  ..
}
```